core/src/main/java/com/github/games647/fastlogin/core/shared/FloodgateManagement.java

```diff
@@ -76,11 +76,38 @@ public void run() {
 
         //this happens on Bukkit if it's connected to Bungee
         //if that's the case, players will be logged in via plugin messages
-        if (core.getStorage() == null) {
+        if (core.getAuthStorage() == null) {
             return;
         }
 
-        profile = core.getStorage().loadProfile(username);
+        profile = core.getAuthStorage().loadProfile(username);
+        if (!profile.isSaved()) {
+            // linked players are stored as Java (= not Floodgate) players
+            profile.setFloodgate(!isLinked);
+        } else {
+            if (!profile.isFloodgateMigrated()) {
```

Comment:
Please put the non-inverted case first. It's easier to follow the flow.

Reply:
Resolved in 35c289d26d115ffaaafdc02dc09f36f1b02ce175